source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2397 @@
+* ``blockTimeMS``: Optional integer, which is required when ``blockConnection``
+  is ``true``. The number of milliseconds for which the affected commands should
+  be blocked. New in server 4.3.4
+  (`SERVER-41070 <https://jira.mongodb.org/browse/SERVER-41070>`_).
+
+
+Determining if a Sharded Cluster Uses Replica Sets
+--------------------------------------------------
+
+When connected to a mongos server, the test runner can query the
+`config.shards <https://docs.mongodb.com/manual/reference/config-database/#config.shards>`__
+collection. Each shard in the cluster is represented by a document in this
+collection. If the shard is backed by a single server, the ``host`` field will
+contain a single host. If the shard is backed by a replica set, the ``host``
+field contain the name of the replica followed by a forward slash and a
+comma-delimited list of hosts.
+
+
+Design Rationale
+================
+
+This specification was primarily derived from the test formats used by the
+`Transactions <../transactions/transactions.rst>`__ and
+`CRUD <../crud/crud.rst>`__ specs, which have served models or other specs.
+
+
+Breaking Changes
+================
+
+This section is reserved for future use. Any breaking changes to the test format
+should be described here in detail for historical reference, in addition to any
+shorter description that may be added to the `Change Log`_.
+
+
+Open Questions
+==============
+
+Note: these questions should be resolved and this section removed before
+publishing version 1.0 of the spec.
+
+
+General
+-------
+
+
+Interaction between top-level and test-level runOn
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are at least two ways to handle runOn requirements defined at both the
+test and file level:
+
+* If `test.runOn`_ is specified, those requirements are checked in addition to
+  any top-level `runOn`_. This behavior is the most straightforward as test
+  runners will not have to consider inheritance or overriding anything (similar
+  to reasons top-level ``databaseName`` and ``collectionName`` fields were
+  ultimately removed). If the top-level `runOn`_ fails, the test runner can skip
+  the entire file outright (i.e. short-circuit the logical-and of both checks).
+
+* If `test.runOn`_ is specified, those requirements are checked *instead* of any
+  top-level `runOn`_. This would mean that a test runner cannot immediately skip
+  an entire file based on evaluation of the top-level `runOn`_; however, it also
+  means that humans editing a test file can no longer trust that the top-level
+  `runOn`_ applies to all tests in the file.
+
+During review, it was asked if a test runner could enforce that `test.runOn`_ is
+only more restrictive (and never more permissive) than the top-level `runOn`_.
+While this is technically feasible, it would add considerable complexity to a
+test runner to make such comparisons.
+
+
+Mixing event types in expectedEvent
```

Comment:
I think to keep things simple, we should defer this to a later version of this unified test format.

Reply:
Moved to Future Work.